jax/lax_linalg.py

```diff
@@ -316,3 +323,49 @@ def qr_jvp_rule(primals, tangents, full_matrices):
 qr_p.def_abstract_eval(qr_abstract_eval)
 xla.translations[qr_p] = qr_translation_rule
 ad.primitive_jvps[qr_p] = qr_jvp_rule
+
+
+# SVD decomposition
+
+def svd_impl(operand, full_matrices, compute_uv):
+  s, u, vt = xla.apply_primitive(svd_p, operand, full_matrices=full_matrices, compute_uv=compute_uv)
+  return core.pack((s, u, vt))
+
+def svd_translation_rule(c, operand, full_matrices, compute_uv):
+  raise NotImplementedError(
+    "Singular value decomposition is only implemented on the CPU backend")
+
+def svd_abstract_eval(operand, full_matrices, compute_uv):
+  if isinstance(operand, ShapedArray):
+    if operand.ndim < 2:
+      raise ValueError("Argument to singular value decomposition must have ndims >= 2")
+
+    batch_dims = operand.shape[:-2]
+    m = operand.shape[-2]
+    n = operand.shape[-1]
+    s = ShapedArray(batch_dims + (min(m, n),), operand.dtype)
+    u = ShapedArray(batch_dims + (m, m if full_matrices else min(m, n)), operand.dtype)
+    vt = ShapedArray(batch_dims + (n if full_matrices else min(m, n), n), operand.dtype)
+  else:
+    s = operand
+    u = operand
+    vt = operand
+  return core.AbstractTuple((s, u, vt))
+
+def svd_cpu_translation_rule(c, operand, full_matrices, compute_uv):
+  shape = c.GetShape(operand)
+  dtype = shape.element_type().type
+  if len(shape.dimensions()) == 2 and dtype in {np.float32, np.float64}:
```

Comment:
You might also consider implementing the complex64 case, but it's not mandatory for a first version. In general I'm trying to make sure we have good complex number support for linear algebra ops.

Reply:
I could send that in a separate PR, if you'd like it that way. I faced some aborted errors, so I have kept that on hold tentatively. I will work on that once this is good to merge.